extensions/hibernate-orm/deployment/src/main/java/io/quarkus/hibernate/orm/deployment/integration/QuarkusClassFileLocator.java

```diff
@@ -0,0 +1,43 @@
+package io.quarkus.hibernate.orm.deployment.integration;
+
+import java.io.IOException;
+
+import io.quarkus.deployment.util.IoUtil;
+import net.bytebuddy.dynamic.ClassFileLocator;
+
+/**
+ * Custom implementation of a ClassFileLocator which will load resources
+ * from the context classloader which is set at the time of the locate()
+ * operation is being performed.
+ * Using a regular ForClassLoader implementation would capture the currently
+ * set ClassLoader and keep a reference to it, while we need it to look
+ * for a fresh copy during the enhancement.
+ * Additionally, we might be able to optimize how the resource is actually
+ * being loaded as we control the ClassLoader implementations
+ * (Such further optimisations are not implemented yet).
```

Comment:
I think it's implemented now, isn't it what this `IoUtil` is about?

Reply:
Not quite..  `IoUtil` delegates to the stream implementation, and some of them are very inefficient especially when loading a stram from a compressed jar. Luckily an ORM entity is not frequently stored in jars.

That's where my other classloader changes come into play.. but not all pieces have come together yet.